file-validator.py

```diff
@@ -0,0 +1,239 @@
+from datetime import date
+from pydantic import BaseModel
+from typing import List
+import yaml
+import json
+import os
+from azure.storage.blob import BlobServiceClient
+from PIL import Image
+
+def fill_index_system(filename):
+     if is_json_empty(filename):
+        with open(filename, 'w') as json_file:
+            json.dump(default_json_data, json_file, indent=2)
+
+class ClientData(BaseModel):
+    clientEmail: str
+    clientExpertise: str
+
+class SeedData(BaseModel):
+    seedID: int
+    seedFamily: str
+    seedGenus: str
+    seedSpecies: str
+
+class ImageDataindex(BaseModel):
+    numberOfImages: int
+    
+class AuditTrail(BaseModel):
+    uploadDate: date
+    editedBy: str
+    editDate: date
+    changeLog: str
+    accessLog: str
+    privacyFlag: bool 
+
+class Index(BaseModel):
+    clientData: ClientData
+    imageData: ImageDataindex
+    seedData: SeedData
+    
+class PIndex(BaseModel):
+    clientData: ClientData
+    imageData: ImageDataindex
+    seedData: SeedData
+    auditTrail: AuditTrail
+    
+class UserData(BaseModel):
+    description: str
+    numberOfSeeds: int
+    zoom: float    
+    
+class Picture(BaseModel):
+    userData: UserData 
+    
+class ClientFeedback(BaseModel):
+    correctIdentification: bool
+    historicalComparison: str
+    
+    
+def is_yaml_file(file_path):
+    _, file_extension = os.path.splitext(file_path)
+    return file_extension.lower() == '.yaml'
```

Comment:
yml is also a very common extension

more reliable to look at headers than extension to determine file types:

https://github.com/ahupp/python-magic